data-prepper-core/src/main/java/org/opensearch/dataprepper/plugin/ClasspathPluginProvider.java

```diff
@@ -58,16 +60,31 @@ private Map<String, Map<Class<?>, Class<?>>> scanForPlugins() {
 
         final Map<String, Map<Class<?>, Class<?>>> pluginsMap = new HashMap<>(dataPrepperPluginClasses.size());
         for (final Class<?> concretePluginClass : dataPrepperPluginClasses) {
-            final DataPrepperPlugin dataPrepperPluginAnnotation = concretePluginClass
-                    .getAnnotation(DataPrepperPlugin.class);
+            final DataPrepperPlugin dataPrepperPluginAnnotation = concretePluginClass.getAnnotation(DataPrepperPlugin.class);
             final String pluginName = dataPrepperPluginAnnotation.name();
             final Class<?> supportedType = dataPrepperPluginAnnotation.pluginType();
 
             final Map<Class<?>, Class<?>> supportTypeToPluginTypeMap =
                     pluginsMap.computeIfAbsent(pluginName, k -> new HashMap<>());
             supportTypeToPluginTypeMap.put(supportedType, concretePluginClass);
+
+            checkAndAddIfDeprecatedPluginNameIsNotDefault(pluginsMap, concretePluginClass);
         }
 
         return pluginsMap;
     }
+
+    private void checkAndAddIfDeprecatedPluginNameIsNotDefault(
```

Comment:
You could probably just name this to `addOptionalDeprecatedPluginName` or another method name tied to the intent rather than the approach.